Core/src/Core/Filters/CalculateArrayHistogramFilter.cpp

```diff
@@ -0,0 +1,156 @@
+  auto pSelectedArrayPathsValue = filterArgs.value<MultiArraySelectionParameter::ValueType>(k_SelectedArrayPaths_Key);
+  auto pNewDataGroupNameValue = filterArgs.value<DataPath>(k_NewDataGroupName_Key); // sanity check if is Attribute matrix after impending complex update
+  auto pHistogramSuffix = filterArgs.value<std::string>(k_HistoName_Key);
+
+  PreflightResult preflightResult;
+
+  complex::Result<OutputActions> resultOutputActions;
+
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  if(pNewDataGroupValue)
+  {
+    auto createDataGroupAction = std::make_unique<CreateDataGroupAction>(pNewDataGroupNameValue);
+    resultOutputActions.value().actions.push_back(std::move(createDataGroupAction));
+  }
+  for(auto& selectedArrayPath : pSelectedArrayPathsValue)
+  {
+    const auto& dataArray = dataStructure.getDataAs<IDataArray>(selectedArrayPath);
+    DataPath childPath;
+    if(pNewDataGroupValue)
+    {
+      childPath = pNewDataGroupNameValue.createChildPath((dataArray->getName() + pHistogramSuffix));
+    }
+    else
+    {
+      childPath = pDataGroupNameValue.createChildPath((dataArray->getName() + pHistogramSuffix));
+    }
+    auto createArrayAction = std::make_unique<CreateArrayAction>(complex::DataType::float64, std::vector<usize>{static_cast<usize>(pNumberOfBinsValue)}, std::vector<usize>{2}, childPath);
+    resultOutputActions.value().actions.push_back(std::move(createArrayAction));
+  }
+
+  return {std::move(resultOutputActions), std::move(preflightUpdatedValues)};
+}
+
+//------------------------------------------------------------------------------
+Result<> CalculateArrayHistogramFilter::executeImpl(DataStructure& dataStructure, const Arguments& filterArgs, const PipelineFilter* pipelineNode, const MessageHandler& messageHandler,
+                                                    const std::atomic_bool& shouldCancel) const
+{
+  CalculateArrayHistogramInputValues inputValues;
+
+  inputValues.NumberOfBins = filterArgs.value<int32>(k_NumberOfBins_Key);
+  inputValues.UserDefinedRange = filterArgs.value<bool>(k_UserDefinedRange_Key);
+  inputValues.MinRange = filterArgs.value<float64>(k_MinRange_Key);
+  inputValues.MaxRange = filterArgs.value<float64>(k_MaxRange_Key);
+  inputValues.SelectedArrayPaths = filterArgs.value<MultiArraySelectionParameter::ValueType>(k_SelectedArrayPaths_Key);
+
+  DataPath dataGroupPath;
+  if(filterArgs.value<bool>(k_NewDataGroup_Key))
+  {
+    dataGroupPath = filterArgs.value<DataPath>(k_NewDataGroupName_Key);
+  }
+  else
+  {
+    dataGroupPath = filterArgs.value<DataPath>(k_DataGroupName_Key);
+  }
+  std::vector<DataPath> createdDataPaths;
+  for(auto& selectedArrayPath : inputValues.SelectedArrayPaths) // regenerate based on preflight
+  {
+    const auto& dataArray = dataStructure.getDataAs<IDataArray>(selectedArrayPath);
+    auto childPath = dataGroupPath.createChildPath((dataArray->getName() + "Histogram"));
```

Comment:
The path being generated here is not the same as the path being generated in the preflight method which then leads to crashes. (Look at lines 110 and 114)